pennylane/labs/resource_estimation/templates/stateprep.py

```diff
@@ -0,0 +1,595 @@
+        else:
+            raise TypeError(f"`selswap_depths` must be an integer, None or iterable")
+
+        params = {
+            "num_state_qubits": num_state_qubits,
+            "precision": precision,
+            "positive_and_real": positive_and_real,
+            "selswap_depths": selswap_depths,
+        }
+        return CompressedResourceOp(cls, params)
+
+    @classmethod
+    def _default_decomp_logic(
+        cls,
+        use_phase_grad_trick,
+        num_state_qubits,
+        positive_and_real,
+        precision=None,
+        selswap_depths=1,
+        **kwargs,
+    ):
+        r"""Returns a list representing the resources of the operator. Each object in the list
+        represents a gate and the number of times it occurs in the circuit.
+
+        Args:
+            use_phase_grad_trick (bool): a flag which determines if the phase gradient trick is used
+                instead of controlled-RY gates and phaseshifts.
+            num_state_qubits (int): number of qubits required to represent the state-vector
+            positive_and_real (bool): Flag that the coefficients of the statevector are all real
+                and positive.
+            precision (float): The precision threshold for loading in the binary representation
+                of the rotation angles.
+            select_swap_depths (Union[None, int, Iterable(int)], optional): A parameter of :code:`QROM`
+                used to trade-off extra qubits for reduced circuit depth.
+
+        Resources:
+            The resources for QROMStatePreparation are according to the decomposition as described
+            in `arXiv:0208112 <https://arxiv.org/abs/quant-ph/0208112>`_, using
+            :class: `~.labs.resource_estimation.ResourceQROM` to dynamically load the rotation angles.
+
+            Controlled-RY (and phase shifts) gates are used to apply all of the rotations coherently. If
+            :code:`use_phase_grad_trick == True` then these rotations gates are implmented using an
+            inplace controlled semi-adder operation (see figure 4. of
+            `arXiv:2409.07332 <https://arxiv.org/pdf/2409.07332>`)_.
+
+        Returns:
+            list[GateCount]: A list of GateCount objects, where each object
+            represents a specific quantum gate and the number of times it appears
+            in the decomposition.
+        """
+        gate_counts = []
+        precision = precision or kwargs["config"]["precision_qrom_state_prep"]
+
+        expected_size = num_state_qubits if positive_and_real else num_state_qubits + 1
+        if isinstance(selswap_depths, int) or selswap_depths is None:
+            selswap_depths = [selswap_depths] * expected_size
+
+        num_precision_wires = math.ceil(math.log2(math.pi / precision)) + 1
+        gate_counts.append(AllocWires(num_precision_wires))
+
+        for j in range(num_state_qubits):
+            num_bitstrings = 2**j
+            num_bit_flips = max(2 ** (j - 1), 1)
```

Comment:
are these the target flips? if that is the case, maybe the length of the bit string is important

Reply:
They are the target flips, I guess it depends on the value of the phase angle right?

Reply:
Yes let me add this back in, I am not sure why I thought it could be removed in this case.